src/components/application_manager/src/resumption/resumption_data_processor.cc

```diff
@@ -68,13 +68,27 @@ void ResumptionDataProcessor::Restore(ApplicationSharedPtr application,
   SetGlobalProperties(application, saved_app);
   AddSubscriptions(application, saved_app);
   register_callbacks_[application->app_id()] = callback;
+
+  if (HasNoHMIRequestsSent(application->app_id())) {
+    LOG4CXX_DEBUG(logger_,
+                  "No HMI requests sent, but resumption is successful");
+    callback(mobile_apis::Result::SUCCESS, "Data resumption succesful");
+    return;
+  }
 }
 
 bool ResumptionRequestIDs::operator<(const ResumptionRequestIDs& other) const {
   return correlation_id < other.correlation_id ||
          function_id < other.function_id;
 }
 
+bool ResumptionDataProcessor::HasNoHMIRequestsSent(const int32_t app_id) {
+  return ((resumption_status_.find(app_id) == resumption_status_.end())) ||
```

Comment:
@LitvinenkoIra Could you be so kind to divide this complex return statement into at least 2 const bool values?